### PR TITLE
Fix typings for Flatbuffers v1.10+ release

### PR DESCRIPTION
**Pull request by rgiese**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/google/flatbuffers/pull/5326
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/flatbuffers/index.d.ts [hunk 1]
```diff
@@ -167,7 +169,8 @@ declare global {
        */
       endObject(): Offset;
 
-      finish(root_table: Offset, file_identifier?: string): void;
+      finish(root_table: Offset, file_identifier?: string, size_prefix?: boolean): void;
```
Comment (peterblazejewicz) [inline types/flatbuffers/index.d.ts]: as you ticket the test options, can we have the test for the method udpated, e.g. like:
```ts
  static finishMonsterBuffer(builder: flatbuffers.Builder, offset: flatbuffers.Offset) {
    builder.finish(offset);
    builder.finish(offset, 'MONS');
    builder.finish(offset, 'MONS', true);
  }
```

File: types/flatbuffers/index.d.ts [hunk 2]
```diff
@@ -167,7 +169,8 @@ declare global {
...
+      finishSizePrefixed(root_table: Offset, file_identifier?: string): void;
```
Comment (peterblazejewicz) [inline types/flatbuffers/index.d.ts]: same here, that's new, would be nice to have the method added to tests as well.

Comment (rgiese) [reply types/flatbuffers/index.d.ts]: Done. Keep in mind these tests don't appear to be _real_ tests, they're just making sure it all compiles.

**Review by peterblazejewicz — APPROVED** [review]
LGTM
@rgiese  thanks for contribution!

**Comment by aardappel** [comment]
@rgiese awesome! was just running into these mismatches elsewhere :)

**Pull request merged**